cns/restserver/util.go

```diff
@@ -160,21 +155,157 @@ func (service *HTTPRestService) saveNetworkContainerGoalState(req cns.CreateNetw
 			service.state.ContainerIDByOrchestratorContext[podInfo.PodName+podInfo.PodNamespace] = req.NetworkContainerid
 			break
 
+		case cns.KubernetesCRD:
+			// Validate and Update the SecondaryIpConfig state
+			returnCode, returnMesage := service.updateIpConfigsStateUntransacted(req, existingSecondaryIPConfigs)
+			if returnCode != 0 {
+				return returnCode, returnMesage
+			}
 		default:
 			errMsg := fmt.Sprintf("Unsupported orchestrator type: %s", service.state.OrchestratorType)
 			logger.Errorf(errMsg)
 			return UnsupportedOrchestratorType, errMsg
 		}
+
 	default:
 		errMsg := fmt.Sprintf("Unsupported network container type %s", req.NetworkContainerType)
 		logger.Errorf(errMsg)
 		return UnsupportedNetworkContainerType, errMsg
 	}
 
+	service.state.ContainerStatus[req.NetworkContainerid] =
+		containerstatus{
+			ID:                            req.NetworkContainerid,
+			VMVersion:                     req.Version,
+			CreateNetworkContainerRequest: req,
+			HostVersion:                   hostVersion}
+
 	service.saveState()
 	return 0, ""
 }
 
+// This func will compute the deltaIpConfigState which needs to be updated (Added or Deleted)
+// from the inmemory map
+// Note: Also this func is an untransacted API as the caller will take a Service lock
+func (service *HTTPRestService) updateIpConfigsStateUntransacted(req cns.CreateNetworkContainerRequest, existingSecondaryIPConfigs map[string]cns.SecondaryIPConfig) (int, string) {
+	// parse the existingSecondaryIpConfigState to find the deleted Ips
+	newIPConfigs := req.SecondaryIPConfigs
+	var tobeDeletedIpConfigs = make(map[string]cns.SecondaryIPConfig)
+
+	// Populate the ToBeDeleted list, Secondary IPs which doesnt exist in New request anymore.
+	// We will later remove them from the in-memory cache
+	if existingSecondaryIPConfigs != nil {
```

Comment:
FYI There is a difference between an unallocated map (`nil`) and an allocated but empty map. A len check will return 0 for both, thus might be a slightly better check for your purposes. At the same time, `range` will basically be a no-op for both cases.

Reply:
okay, then i will remove the nil check. Thanks for letting me know.